### PR TITLE
refractor: differentiate between file existence and network errors

### LINKED ISSUE #502: ERROR: You seem to been offline, but Im clearly not :/
Captura de tela de 2022-02-13 11-30-52
Captura de tela de 2022-02-13 11-38-42

### PR DESCRIPTION
**Pull request by Henryws**
## Purpose

Shows the difference between a broken link, or no network.

## Approach

Curl the header first, and check the error code.

## Addendum

Fixes #502

## Checklist

- [x] I confirm that I have read the [contributing guidelines](https://github.com/pacstall/pacstall/blob/develop/CONTRIBUTING.md), and this pull request is abiding by all the clauses stated in the guideline.

### CODE REVIEW

File: misc/scripts/download.sh [hunk 1]
```diff
@@ -24,25 +24,34 @@
 
 # This script downloads pacscripts from the interwebs
 
-if curl --output /dev/null --silent --head --fail -- "$URL" ; then
+error_code="$(curl -o /dev/null -s -w "%{http_code}\n" -- "${URL}")"
+if [[ "${error_code}" != "200" ]]; then
```
Comment (D-Brox) [inline misc/scripts/download.sh:28]: Call it http code, not error code.

File: misc/scripts/download.sh [hunk 2]
```diff
@@ -24,25 +24,34 @@
...
 
-if curl --output /dev/null --silent --head --fail -- "$URL" ; then
+http_code="$(curl -o /dev/null -s -w "%{http_code}\n" -- "${URL}")"
+if [[ "${http_code}" != "200" ]]; then
+	if [[ "${http_code}" == "000" ]]; then
+		fancy_message error "Failed to download file, check your internet connection"
+		exit 1
+	fi
+	fancy_message error "Package does not exist"
+	exit 1
```
Comment (wizard-28) [inline misc/scripts/download.sh:34]: Do this on `404` specifically, and for other non-`200` http codes print an error message with the http code.

**Review by wizard-28 — CHANGES_REQUESTED** [review]

**Review by wizard-28 — APPROVED** [review]
Looks good to me, waiting on @D-Brox

File: misc/scripts/search.sh [hunk 3]
```diff
@@ -75,6 +75,9 @@ fi
 PACKAGELIST=()
 URLLIST=()
 while IFS= read -r URL; do
+	if ! check_url "${URL}/packagelist"; then
+		exit 1
```
Comment (D-Brox) [inline misc/scripts/search.sh:79]: Add warning for the invalid repo

**Review by D-Brox — CHANGES_REQUESTED** [review]

File: misc/scripts/search.sh (hunk 3 above)
Comment (Henryws) [reply misc/scripts/search.sh:79]: An invalid repo will return the error code 404.

**Review by D-Brox — APPROVED** [review]

**Pull request merged**